### DISCUSSION TITLE
Checks fail on deployment using second production GitHub organisation

### DISCUSSION THREAD

**Discussion by nikolabul** [open-sdg/open-sdg#1832] [Q&A] [answered]
I wanted to change the domain of my [sdg site](https://github.com/montestat/site) so I followed this [tutorial](https://open-sdg.readthedocs.io/en/latest/hosting/github-pages-production/) to deploy the app using second production GitHub organisation. Everything worked fine until I edited [deploy-to-production.yml](https://github.com/montestat/data/tree/develop/.github/workflows/deploy-to-production.yml) in my [data repository](https://github.com/montestat/data) after which the checks failed.

I get the following error:

```
Run python scripts/build_data.py
  
Traceback (most recent call last):
  File "scripts/build_data.py", line 3, in <module>
    open_sdg_build(config='config_data.yml')
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/open_sdg.py", line 150, in open_sdg_build
    outputs = open_sdg_prep(options)
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/open_sdg.py", line 327, in open_sdg_prep
    indicator_export_filename=options['indicator_export_filename'])
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/outputs/OutputOpenSdg.py", line 33, in __init__
    indicator_options, logging=logging)
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/outputs/OutputBase.py", line 40, in __init__
    self.indicators = self.merge_inputs(inputs)
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/outputs/OutputBase.py", line 114, in merge_inputs
    input.execute_once(self.indicator_options)
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/inputs/InputBase.py", line 51, in execute_once
    self.execute(indicator_options)
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/inputs/InputSdmxMeta.py", line 14, in execute
    indicator_map = self.get_indicator_map()
  File "/opt/hostedtoolcache/Python/3.7.14/x64/lib/python3.7/site-packages/sdg/inputs/InputSdmx.py", line 223, in get_indicator_map
    indicator_names.append(indicator_names[0])
IndexError: list index out of range
Error: Process completed with exit code 1.
```

The checks fail even when I revert the changes made in [deploy-to-production.yml](https://github.com/montestat/data/tree/develop/.github/workflows/deploy-to-production.yml). I can see the errors are linked to indicators (scripts/build_data.py) even if I didn't change anything related to them. 

Could you please suggest what could be done to resolve this issue?

**Comment by LucyGwilliamAdmin** [comment, marked as answer]
Hi @nikolabul, I think you need to add a line at line 37 with 'persist_credentials: false' e.g. like in data starter repo: https://github.com/open-sdg/open-sdg-data-starter/blob/develop/.github/workflows/deploy-to-production.yml#L37

The file has changed slightly since you first set up your repos from the template in order to counteract the issue you are having.

Let me know if this works

**Reply by nikolabul** [reply]
Hi @LucyGwilliamAdmin, thanks for a quick reply. I added  'persist_credentials: false' to deploy-to-production.yml but the tests fail again.

**Reply by brockfanning** [reply]
@nikolabul Can you also try upgrading this line to version 2.0.2? https://github.com/open-sdg/open-sdg-data-starter/blob/develop/scripts/requirements.txt#L2 (edit - typed the version wrong, should be 2.0.2)

**Reply by nikolabul** [reply]
@brockfanning thanks, that upgrade resolved the issues.

**Comment by nikolabul** [comment]
Hi @brockfanning. The production platform is up and running but I have additional issue now.

The staging website loads the content fine:
[https://montestat.github.io/site/en/1-1-1/](https://montestat.github.io/site/en/1-1-1/)

The production website doesn't load the chart/table and the filters (there are just the loading circles):
[https://www.sdgmontenegro.me/en/1-1-1/](https://www.sdgmontenegro.me/en/1-1-1/)

What could be the fix for this issue?

**Reply by brockfanning** [reply]
@nikolabul It's possible that this is related to the "remote_data_prefix" not using the custom domain. We just recently added some clarification about that in the documentation, which you can see here: https://github.com/open-sdg/open-sdg/blob/2.2.0-dev/docs/hosting/github-pages-production.md#custom-domain

**Reply by nikolabul** [reply]
Thanks a lot @brockfanning. It works now ([https://www.sdgmontenegro.me/en/1-1-1/](https://www.sdgmontenegro.me/en/1-1-1/)).

I found this explanation in the documentation you sent me:
> Note that if your production data is under the same Github organization as your production site, then after you set up the custom domain you will need to change the remote_data_prefix line in your production site configuration file such that it uses the custom domain.

`remote_data_prefix: https://my-custom-domain.org/my-data-repo`

I suggest that you make a change in the documentation as the 'remote_data_prefix' should be set in staging site configuration instead of production site configuration.
